inspire_mitmproxy/services/base_service.py

```diff
@@ -57,10 +60,19 @@ def process_request(self, request: MITMRequest) -> MITMResponse:
             if interaction.matches_request(request):
                 response = interaction.response
                 interaction.execute_callbacks()
+                self.increment_interaction_count(interaction.name)
                 return response
 
         raise NoMatchingRecording(self.name, request)
 
+    def increment_interaction_count(self, interaction_name: str):
+        try:
+            self.interactions_replayed[interaction_name]['num_calls'] += 1
+        except KeyError:
+            self.interactions_replayed[interaction_name] = {
+                'num_calls': 1,
+            }
+
```

Comment:
If you have other keys in `self.interactions_replayed[interaction_name]` you will lose them, since you are assigning a new dictionary to it. I'd use `self.interactions_replayed[interaction_name]['num_calls'] = 1` instead.